utils/ansible-playbooks/deploy-monitoring/templates/ocp-monitor.j2

```diff
@@ -41,6 +41,63 @@ bash_backtrace() {
     echo "Exiting with status ${RET}"
     exit "${RET}"
 }
+
+## FUNCTIONS FOR STATE MACHINE CHECKS
+
+function get_state () {
+# Parameters passed:
+#  - unique name identifying what is being tracked state-wise
+#  - variable to hold the current state, either read from state file or initialized when not yet set.
+# Also forms the basis for the file-name used to track state.
+# If the file does not exist, then a state of 0 is assigned and the state file created.
+
+  local STATENAME="$1"
+  local STATEVALUE="$2"
+  local CURRENT_STATE=0
+
+  STATE_FILE="/tmp/ocp-monitor."${STATENAME}".state"
+
+  if [ -f "${STATE_FILE}" ]; then
+    CURRENT_STATE=`cat "${STATE_FILE}"`;
+  else
+    CURRENT_STATE=`echo 0 | tee "${STATE_FILE}"`
+  fi
+  eval "${STATEVALUE}"="${CURRENT_STATE}"
```

Comment:
To prevent me from becoming confused in the future about eval, can we add some comments where its used?

> Set the global variable named by the string in $STATEVALUE to the value in $CURRENT_STATE
> ie: NODE_READY_STATE=0